projects/ideabox-groups.md

```diff
@@ -0,0 +1,210 @@
+* The page _should not_ reload when an idea is deleted.
+* The idea should be removed from `localStorage`. It should not re-appear on next page load.
+  * This update of the data model should happen in a `deleteFromStorage` method that is defined in the `Idea` class.
+  * How the dom is updated is up to you.
+
+#### Editing an existing idea
+
+* When a user clicks the title or body of an idea in the list, that text should become an editable text field, pre-populated with the existing idea title or body.
+* The user should be able to "commit" their changes by pressing "Enter/Return" or by clicking outside of the text field.
+* If the user reloads the page, their edits will be reflected.
+  * This update of the data model should occur in an `updateSelf` method that is defined in the `Idea` class.
+  * How the dom is updated is up to you.
+
+#### Changing the quality of an idea
+
+As we said above, ideas should start out as "swill." In order to change the recorded quality of an idea, the user will interact with it from the idea list.
+
+* Each idea in the list should include an "upvote" and "downvote" button.
+* Clicking upvote on the idea should increase its quality one notch ("swill" → "plausible",
+  "plausible" → "genius").
+* Clicking downvote on the idea should decrease its quality one notch ("genius" → "plausible",
+  "plausible" → "swill").
+* Incrementing a "genius" idea or decrementing a "swill" idea should have no effect.
+  * This update of the data model should occur in an `updateQuality` method that is defined in the `Idea` class.
+  * How the dom is updated is up to you.
+
+#### Filtering and Searching by Text
+
+We'd like our users to be able to easily find specific ideas they already created, so let's provide them with a filtering interface on the idea list.
+
+* At the top of the idea list, include a text field labeled "Search".
+* As a user types in the search box, the list of ideas should filter in real time to only display ideas whose title or body include the user's text. The page _should not_ reload.
+* Clearing the search box should restore all the ideas to the list.
+  * There is no need to make persisting changes to the data model to achieve this functionality.
+
+#### Filter by Importance
+
+The application should allow users to filter the idea list based on level of quality.
+
+* Your application should have 3 buttons corresponding to each level of importance (swill, plausible, and genius).
+* When one of the filter buttons is clicked, the TODO list should only display TODOs with the selected importance.
+  * There is no need to make persisting changes to the data model to achieve this functionality.
+
+#### Recent Ideas
+
+The application should only show the ten most recent Ideas on page load.
+
+* The application should contain a button labeled `Show more ideas...`.
+* When a user clicks on the `Show more TODOs...` button, this list should load additional ideas from the past.
```

Comment:
You have a `Show more ideas` and a `Show more TODOs` in both bullet points for the button. Oops. 😉 The other thing I would clarify, is the point about loading additional ideas from the past.  If I click the button, should it load 10 more ideas?  Or should it load the rest of the ideas?

Reply:
lolol it should show the rest of the ideas... good catch